tests/Kernel/StatusMessagesTest.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme\Kernel;
+
+use Drupal\Core\Messenger\MessengerInterface;
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Tests the rendering of the status messages.
+ */
+class StatusMessagesTest extends AbstractKernelTest {
+
+  /**
+   * The message types available.
+   *
+   * Each element contains:
+   *   - heading: the heading shown in the interface.
+   *   - modifier: the string appended as class modifier to the wrapper of the
+   *     message of a certain type.
+   *
+   * @var array
+   */
+  const MESSAGE_TYPES = [
+    MessengerInterface::TYPE_STATUS => [
+      'heading' => 'Status message',
+      'modifier' => 'success',
+    ],
+    MessengerInterface::TYPE_WARNING => [
+      'heading' => 'Warning message',
+      'modifier' => 'warning',
+    ],
+    MessengerInterface::TYPE_ERROR => [
+      'heading' => 'Error message',
+      'modifier' => 'error',
+    ],
+  ];
+
+  /**
+   * Tests the rendering of the status messages.
+   *
+   * @param array $data
+   *   An array of messages, keyed by message type.
+   *
+   * @dataProvider statusMessagesProvider
+   */
+  public function testStatusMessages(array $data): void {
+    $messenger = \Drupal::messenger();
+
+    foreach ($data as $type => $messages) {
+      foreach ($messages as $message) {
+        $messenger->addMessage($message, $type);
+      }
+    }
+
+    $render = ['#type' => 'status_messages'];
+    $html = $this->renderRoot($render);
+    $crawler = new Crawler($html);
+
+    foreach ($data as $type => $messages) {
+      $modifier = self::MESSAGE_TYPES[$type]['modifier'];
+      $heading = self::MESSAGE_TYPES[$type]['heading'];
+
+      $wrapper = $crawler->filter('div.ecl-message--' . $modifier);
+      $this->assertCount(1, $wrapper, sprintf('Wrong number of wrappers found for "%s" messages.', $type));
+
+      $title = $wrapper->filter('div.ecl-message__title');
+      $this->assertCount(1, $title, sprintf('Wrong number of headings found for "%s" messages.', $type));
```

Comment:
This is nice, could we port this to the other tests as well? Like now we do:

```
    // Assert list contains ECL classes.
    $actual = $crawler->filter('ul.ecl-navigation-list.ecl-navigation-list--tabs')->count();
    $this->assertEquals(1, $actual);
```
While it could actually simply be
```
    $this->assertCount(1, $actual);
```

Reply:
Comment was adressed in the last commit.